server/app/services/report_renderer.py

```
@@ -0,0 +1,410 @@
+        organization_name: str,
+        rooftop_name: Optional[str] = None,
+        logo_svg: Optional[str] = None,
+    ) -> str:
+        """
+        Build a print-optimized HTML string from the report data.
+        """
+        logo_markup = logo_svg or ReportRenderer._load_logo_svg(prefer_full=True)
+        period = report.get("period", {})
+        kpis = report.get("kpis", {})
+        avg_nps = kpis.get("avg_nps")
+        with_nps = kpis.get("with_nps") or 0
```

Ruff (0.14.4)

Local variable `with_nps` is assigned to but never used

Remove assignment to unused variable `with_nps`

(F841)